website/server/libs/taskManager.js

```diff
@@ -281,3 +291,172 @@ export function moveTask (order, taskId, to) {
+    });
+
+    managerPromises.push(task.save());
+    await Promise.all(managerPromises);
+
+    throw new NotAuthorized(res.t('taskApprovalHasBeenRequested'));
+  }
+  let wasCompleted = task.completed;
+
+  let [delta] = shared.ops.scoreTask({task, user, direction}, req);
+  // Drop system (don't run on the client, as it would only be discarded since ops are sent to the API, not the results)
+  if (direction === 'up') shared.fns.randomDrop(user, {task, delta}, req, res.analytics);
+
+  // If a todo was completed or uncompleted move it in or out of the user.tasksOrder.todos list
+  // TODO move to common code?
+  let taskOrderPromise;
+  if (task.type === 'todo') {
+    if (!wasCompleted && task.completed) {
+      // @TODO: mongoose's push and pull should be atomic and help with
+      // our concurrency issues. If not, we need to use this update $pull and $push
+      taskOrderPromise = user.update({
+        $pull: { 'tasksOrder.todos': task._id },
+      }).exec();
+      // user.tasksOrder.todos.pull(task._id);
+    } else if (wasCompleted && !task.completed && user.tasksOrder.todos.indexOf(task._id) === -1) {
+      taskOrderPromise = user.update({
+        $push: { 'tasksOrder.todos': task._id },
+      }).exec();
+      // user.tasksOrder.todos.push(task._id);
+    }
+  }
+
+  setNextDue(task, user);
+
+  if (task.group && task.group.taskId) {
+    await handleSharedCompletion(task);
+  }
+
+  taskScoredWebhook.send(user, {
+    task,
+    direction,
+    delta,
+    user,
+  });
+
+  // Track when new users (first 7 days) score tasks
+  if (moment().diff(user.auth.timestamps.created, 'days') < 7) {
+    res.analytics.track('task score', {
+      uuid: user._id,
+      hitType: 'event',
+      category: 'behavior',
+      taskType: task.type,
+      direction,
+      headers: req.headers,
+    });
+  }
+  return {user, task, delta, direction, taskOrderPromise};
+}
+
+async function handleChallengeTask (task, delta, direction, resolve, reject) {
+  if (task.challenge && task.challenge.id && task.challenge.taskId && !task.challenge.broken && task.type !== 'reward') {
+    // Wrapping everything in a try/catch block because if an error occurs using `await` it MUST NOT bubble up because the request has already been handled
+    try {
+      const chalTask = await Tasks.Task.findOne({
+        _id: task.challenge.taskId,
+      }).exec();
+
+      if (!chalTask) return;
+
+      await chalTask.scoreChallengeTask(delta, direction);
+    } catch (e) {
+      logger.error(e);
+      reject();
+    }
+  }
+  resolve();
+}
+
+export async function scoreTasks (user, taskScorings, req, res) {
+  let taskIds = taskScorings.map(taskScoring => taskScoring.id).filter(id => id !== undefined);
+  if (taskIds.length === 0) throw new BadRequest();
+  let tasks = {};
+  (await Tasks.Task.findMultipleByIdOrAlias(taskIds, user._id)).forEach(task => {
+    tasks[task._id] = task;
+    if (task.alias !== undefined && task.alias.length > 0) {
```

Comment:
I'd just check for `if (task.alias)` it covers all the falsey values including undefined and empty strings